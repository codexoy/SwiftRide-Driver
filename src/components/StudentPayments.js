import React, { Fragment } from "react"
import { StyleSheet, View, Text, FlatList } from "react-native"
import Separator from "./Separator"
import Row from "./Row"
import moment from "moment"
import { DATE_FORMAT, colors } from "../consts"
import EmptyState from "./EmptyState"
import { strings } from "../i18n"
import PaymentsLoader from "./PaymentsLoader"
import ShowReceipt from "./ShowReceipt"

export default class StudentPayment extends React.Component {
	constructor(props) {
		super(props)
	}
	renderPaymentItem = ({ item, index }) => {
		let firstItemStyles = {}
		if (index == 0) {
			firstItemStyles = { marginTop: 0 }
		}
		return (
			<Row
				style={{ ...styles.paymentRow, ...firstItemStyles }}
				leftSide={
					<Text style={styles.amountOfStudent}>{item.amount}₪</Text>
				}
				key={`payment${item.id}`}
			>
				<Text style={styles.dateOfPayment}>
					{moment
						.utc(item.created_at)
						.local()
						.format(DATE_FORMAT)}
				</Text>
				<ShowReceipt
					item={item}
					dispatch={this.props.dispatch}
					user={this.props.user}
					style={{ marginRight: -16 }}
				/>
			</Row>
		)
	}

	_renderEmpty = () => (
		<EmptyState
			image="payments"
			text={strings("empty_payments")}
			imageSize="small"
		/>
	)

	render() {
		if (this.props.loading) {
			return <PaymentsLoader />
		}
		if (this.props.payments.length == 0) {
			return this._renderEmpty()
		}
		let color = colors.green
		if (this.props.sum < 0) color = "red"
		return (
			<Fragment>
				<View style={styles.amountView}>
					<Text style={{ ...styles.amount, color }}>
						{this.props.sum}₪
					</Text>
				</View>
				<Separator />
				<FlatList
					data={this.props.payments}
					renderItem={this.renderPaymentItem}
					keyExtractor={item => `payment${item.id}`}
				/>
			</Fragment>
		)
	}
}

const styles = StyleSheet.create({
	amountView: {
		marginTop: 16,
		alignSelf: "center"
	},
	amount: {
		fontFamily: "Assistant-Light",
		fontSize: 44
	},
	dateOfPayment: {
		fontWeight: "bold"
	},
	paymentRow: {
		marginTop: 12
	}
})
