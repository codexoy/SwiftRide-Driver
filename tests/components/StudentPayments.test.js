import React from "react"

import renderer from "react-test-renderer"

import StudentPayments from "../../src/components/StudentPayments"

const user = { name: "test", teacher_id: 1, is_approved: true }
describe("StudentPayments", () => {
	test("view renders correctly", () => {
		const tree = renderer
			.create(
				<StudentPayments
					user={user}
					sum={1000}
					payments={[
						{
							id: 1,
							amount: 100,
							created_at: new Date(Date.UTC("2019-15-03"))
						},
						{
							id: 2,
							amount: 900,
							created_at: new Date(Date.UTC("2019-15-03"))
						}
					]}
				/>
			)
			.toJSON()
		expect(tree).toMatchSnapshot()
	})
})
