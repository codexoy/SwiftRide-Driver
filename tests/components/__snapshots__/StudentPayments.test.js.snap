// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentPayments view renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "alignSelf": "center",
        "marginTop": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "rgb(24, 199, 20)",
          "fontFamily": "Assistant-Light",
          "fontSize": 44,
        }
      }
    >
      1000
      ₪
    </Text>
  </View>,
  <View
    style={
      Object {
        "borderBottomColor": "rgb(212, 212, 212)",
        "borderBottomWidth": 1,
        "marginBottom": 20,
        "marginTop": 20,
        "width": "100%",
      }
    }
  />,
  <RCTScrollView
    data={
      Array [
        Object {
          "amount": 100,
          "created_at": Date { NaN },
          "id": 1,
        },
        Object {
          "amount": 900,
          "created_at": Date { NaN },
          "id": 2,
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginTop": 0,
              "width": "100%",
            }
          }
        >
          <View
            style={Object {}}
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Invalid date
            </Text>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "marginRight": -16,
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "flex-start",
                    "color": "rgb(12,116,244)",
                    "fontWeight": "bold",
                  }
                }
              >
                צור חשבונית
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "marginLeft": "auto",
              }
            }
          >
            <Text>
              100
              ₪
            </Text>
          </View>
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
              "marginTop": 12,
              "width": "100%",
            }
          }
        >
          <View
            style={Object {}}
          >
            <Text
              style={
                Object {
                  "fontWeight": "bold",
                }
              }
            >
              Invalid date
            </Text>
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "marginRight": -16,
                  "opacity": 1,
                }
              }
            >
              <Text
                style={
                  Object {
                    "alignSelf": "flex-start",
                    "color": "rgb(12,116,244)",
                    "fontWeight": "bold",
                  }
                }
              >
                צור חשבונית
              </Text>
            </View>
          </View>
          <View
            style={
              Object {
                "marginLeft": "auto",
              }
            }
          >
            <Text>
              900
              ₪
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>,
]
`;
